scalapb/src/test/scala/com/linecorp/armeria/server/scalapb/ScalaPbRequestConverterFunctionTest.scala

```diff
@@ -0,0 +1,185 @@
+      ctx.request.headers
+        .withMutations(builder => builder.contentType(MediaType.JSON)),
+      HttpData.ofUtf8(json))
+    val requestObject =
+      converter.convertRequest(
+        ctx,
+        req,
+        typeToken.getRawType,
+        typeToken.getType.asInstanceOf[ParameterizedType])
+    assertThat(requestObject).isEqualTo(collection)
+  }
+
+  @ArgumentsSource(classOf[JsonObjectRequestProvider])
+  @ParameterizedTest
+  def jsonObjectToMap(map: Any, json: String, typeToken: TypeToken[_]): Unit = {
+    val converter = ScalaPbRequestConverterFunction()
+    val req = AggregatedHttpRequest.of(
+      ctx.request.headers
+        .withMutations(builder => builder.contentType(MediaType.JSON)),
+      HttpData.ofUtf8(json))
+    val requestObject =
+      converter.convertRequest(
+        ctx,
+        req,
+        typeToken.getRawType,
+        typeToken.getType.asInstanceOf[ParameterizedType])
+    assertThat(requestObject).isEqualTo(map)
+  }
+
+  @ArgumentsSource(classOf[JsonArrayRequestProvider])
+  @ParameterizedTest
+  def jsonArrayWithNoContentType(collection: Any, json: String, typeToken: TypeToken[_]): Unit = {
+    val converter = ScalaPbRequestConverterFunction()
+    val req = AggregatedHttpRequest.of(ctx.request.headers, HttpData.ofUtf8(json))
+    assertThatThrownBy { () =>
+      converter.convertRequest(
+        ctx,
+        req,
+        typeToken.getRawType,
+        typeToken.getType.asInstanceOf[ParameterizedType])
+    }.isInstanceOf(classOf[FallthroughException])
+  }
+}
+
+private[scalapb] object ScalaPbRequestConverterFunctionTest {
+
+  private val ctx: ServiceRequestContext = ServiceRequestContext.of(HttpRequest.of(HttpMethod.POST, "/"))
+
+  private val simpleRequest1: SimpleRequest = SimpleRequest("Armeria")
+  private val simpleRequest2: SimpleRequest = SimpleRequest("Protobuf")
+  private val printer: Printer = new Printer()
+
+  private class JsonArrayRequestProvider extends ArgumentsProvider {
+    override def provideArguments(context: ExtensionContext): stream.Stream[Arguments] = {
+      val list = List(simpleRequest1, simpleRequest2)
+      val vector = list.toVector
+      val set = list.toSet
+      val jlist = ImmutableList.of(simpleRequest1, simpleRequest2);
+      val jset = ImmutableSet.of(simpleRequest1, simpleRequest2);
+
+      java.util.stream.Stream.of(
+        Arguments.of(list, toJson(list), new TypeToken[List[SimpleRequest]]() {}),
+        Arguments.of(vector, toJson(vector), new TypeToken[Vector[SimpleRequest]]() {}),
+        Arguments.of(set, toJson(set), new TypeToken[Set[SimpleRequest]]() {}),
+        Arguments.of(jlist, toJson(jlist), new TypeToken[java.util.List[SimpleRequest]]() {}),
+        Arguments.of(jset, toJson(jset), new TypeToken[java.util.Set[SimpleRequest]]() {})
+      )
+    }
+
+  }
+
+  private class JsonObjectRequestProvider extends ArgumentsProvider {
+    override def provideArguments(context: ExtensionContext): stream.Stream[Arguments] = {
```

Comment:
We're mixing `stream.Stream` and `java.util.stream.Stream`. Maybe better using a single form?

Reply:
Agreed!